src/components/ContentSidebar/ActivityFeed/Avatar.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @flow
+ * @file avatar component
+ * @author Box
+ */
+import * as React from 'react';
+import AvatarComponent from 'box-react-ui/lib/components/avatar';
+import type { User } from '../../../flowTypes';
+
+type Props = {
+    user: User,
+    getAvatarUrl: (string) => Promise<?string>
+};
+
+type State = {
+    avatarUrl: ?string
+};
+
+class Avatar extends React.PureComponent<Props, State> {
+    state = {
+        avatarUrl: null
+    };
+
+    /**
+     * Success handler for getting avatar url
+     *
+     * @param {string} avatarUrl the user avatar url
+     */
+    getAvatarUrlHandler(avatarUrl: ?string) {
+        this.setState({
+            avatarUrl
+        });
+    }
+
+    getAvatarUrl() {
```

Comment:
jsdocs on functions

Reply:
Also, is this function needed? Can't you just have it in didMount till there is a use case for having it outside?

Reply:
This would be a good candidate for async rendering when its out in later React versions.

Reply:
Updated the docs

Is it really needed to be its own function? No, but it makes it more testable